code/modules/mining/equipment/resonator.dm

```diff
@@ -54,13 +59,22 @@
 	layer = ABOVE_ALL_MOB_LAYER
 	plane = ABOVE_GAME_PLANE
 	duration = 60 SECONDS
+	/// the amount of damage living beings will take whilst inside the field during its burst
 	var/resonance_damage = 20
+	/// the modifier to resonance_damage; affected by the quick_burst_mod from the resonator
 	var/damage_multiplier = 1
+	/// the parent creator (user) of this field
 	var/creator
-	var/obj/item/resonator/res
-	var/rupturing = FALSE //So it won't recurse
-
-/obj/effect/temp_visual/resonance/Initialize(mapload, set_creator, set_resonator, mode)
+	/// the parent resonator of this field
+	var/obj/item/resonator/parent_resonator
+	/// whether the field is rupturing currently or not (to prevent recursion)
+	var/rupturing = FALSE
+	/// the probability that the field will not be able to spread
+	var/failure_prob = 0
+	/// the number that is added to the failure_prob. Will default to 50
+	var/adding_failure
+
+/obj/effect/temp_visual/resonance/Initialize(mapload, set_creator, set_resonator, mode, set_failure = 50)
```

Comment:
I don't really like this optional param, but it's not a deal breaker.

Can you confirm that this call site is correct?  It looks like the parameters might be fucked: https://github.com/tgstation/tgstation/blob/325e5488cd47b83c0ffb65c9a4aca74c9a538e21/code/modules/projectiles/guns/energy/kinetic_accelerator.dm#L485

Reply:
dealt with! It should work fine now.